EOLib/PacketHandlers/Avatar/AvatarReplyHandler.cs

```diff
@@ -0,0 +1,80 @@
+﻿using AutomaticTypeMapper;
+using EOLib.Domain.Character;
+using EOLib.Domain.Extensions;
+using EOLib.Domain.Map;
+using EOLib.Domain.Notifiers;
+using EOLib.Net;
+using EOLib.Net.Handlers;
+using EOLib.Domain.Login;
+using System.Collections.Generic;
+using System;
+
+namespace EOLib.PacketHandlers.Avatar
+{
+    [AutoMappedType]
+    public class AvatarReplyHandler : InGameOnlyPacketHandler
+    {
+        private readonly ICurrentMapStateRepository _currentMapStateRepository;
+        private readonly IEnumerable<IOtherCharacterAnimationNotifier> _otherCharacterAnimationNotifiers;
+        private readonly IEnumerable<IOtherCharacterEventNotifier> _otherCharacterEventNotifiers;
+        private readonly IEnumerable<IMainCharacterEventNotifier> _mainCharacterEventNotifiers;
+        private readonly ICharacterRepository _characterRepository;
+
+        public AvatarReplyHandler(
+            IPlayerInfoProvider playerInfoProvider,
+            ICurrentMapStateRepository currentMapStateRepository,
+            ICharacterRepository characterRepository,
+            IEnumerable<IOtherCharacterAnimationNotifier> otherCharacterAnimationNotifiers,
+            IEnumerable<IMainCharacterEventNotifier> mainCharacterEventNotifiers,
+            IEnumerable<IOtherCharacterEventNotifier> otherCharacterEventNotifiers)
+            : base(playerInfoProvider)
+        {
+            _currentMapStateRepository = currentMapStateRepository;
+            _characterRepository = characterRepository;
+            _otherCharacterAnimationNotifiers = otherCharacterAnimationNotifiers;
+            _mainCharacterEventNotifiers = mainCharacterEventNotifiers;
+            _otherCharacterEventNotifiers = otherCharacterEventNotifiers;
+        }
+
+        public override PacketFamily Family => PacketFamily.Avatar;
+        public override PacketAction Action => PacketAction.Reply;
+        public override bool HandlePacket(IPacket packet)
+        {
+            var attackerId = packet.ReadShort();
+            var victimId = packet.ReadShort();
+            var damage = packet.ReadThree();
+            var direction = (EODirection)packet.ReadChar();
+            var hpPercentage = packet.ReadChar();
+            bool dead = packet.ReadChar() != 0;
+
+            if (victimId == _characterRepository.MainCharacter.ID)
+            {                
+                 _characterRepository.MainCharacter = _characterRepository.MainCharacter.WithDamage(damage, dead);
+
+                foreach (var notifier in _mainCharacterEventNotifiers)
+                {
+                    notifier.NotifyTakeDamage(damage, hpPercentage, isHeal: false);
+                }
+            }
+            
```

Comment:
More whitespace nits -
1) make sure indentation is four spaces per indentation level
2) remove line breaks between if blocks

If I weren't lazy this would be automatically enforced, sorry to have to bug you about it 😭